app/models/user.rb

```diff
@@ -1,5 +1,9 @@
 class User < ApplicationRecord
-  has_many :posts ,dependent: :destroy
-  has_many :comments, dependent: :destroy
-  has_many :likes, dependent: :destroy
+  has_many :posts, foreign_key: :author_id, dependent: :destroy
+  has_many :comments, foreign_key: :author_id, dependent: :destroy
+  has_many :likes, foreign_key: :author_id, dependent: :destroy
+
+  def most_three_recent_post
+    posts.order(created_at: :desc).limit(3)
+  end
```

Comment:
# Optional:
- Can you use the same suggestion I made in models/comments: `after_save :update_likes_counter`